coalib/misc/Decorators.py

```diff
@@ -298,3 +298,24 @@ def decorated(*args, **kwargs):
         return function(*args, **kwargs)
 
     return decorated
+
+
+class classproperty(property):
+    """
+    Decorator to set a class function to a class property.
+
+    >>> class test:
+    ...
+    ...     @classproperty
+    ...     def func(self):
+    ...         return 1
+    ...
+    >>> t = test()
+    >>> t.func
+    1
+    >>> test.func
+    1
```

Comment:
so I'd write a doctest comment usually like:

```
Given a class like:

>>> class ...:
...     def ...():
...         do_something()

If we now access the property of the class:

>>> classname.func
1

And the same behaviour with an instance:

>>> class().func
1
```

(be sure to phrase it out and not use my stupid example texts ;))